src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Helpers.cs

```diff
@@ -196,50 +144,70 @@ private static bool IsAccessible(ISymbol symbol)
             Contract.ThrowIfFalse(equivalentTypesWithDifferingAssemblies.All(kvp => kvp.Key.ContainingType == null));
             Contract.ThrowIfFalse(equivalentTypesWithDifferingAssemblies.All(kvp => kvp.Value.ContainingType == null));
 
-            var referencedAssemblies = new MultiDictionary<string, IAssemblySymbol>();
-            foreach (var assembly in compilation.GetReferencedAssemblySymbols())
+            // Cache compilations so we avoid recreating any as we walk the pairs of types.
+            using var _ = PooledHashSet<Compilation>.GetInstance(out var compilationSet);
+
+            foreach (var (type1, type2) in equivalentTypesWithDifferingAssemblies)
             {
-                referencedAssemblies.Add(assembly.Name, assembly);
+                // Check if type1 was forwarded to type2 in type2's compilation, or if type2 was forwarded to type1 in
+                // type1's compilation.  We check both direction as this API is called from higher level comparison APIs
+                // that are unordered.
+                if (!VerifyForwardedType(solution, type1, type2, compilationSet, cancellationToken) &&
+                    !VerifyForwardedType(solution, type2, type1, compilationSet, cancellationToken))
+                {
+                    return false;
+                }
             }
 
-            var verifiedCount = 0;
-            foreach (var kvp in equivalentTypesWithDifferingAssemblies)
+            return true;
+        }
+
+        /// <summary>
+        /// Returns <see langword="true"/> if <paramref name="type1"/> was forwarded to <paramref name="type2"/> in
+        /// <paramref name="type2"/>'s <see cref="Compilation"/>.
+        /// </summary>
+        private static bool VerifyForwardedType(
+            Solution solution,
+            INamedTypeSymbol type1,
+            INamedTypeSymbol type2,
```

Comment:
```suggestion
            INamedTypeSymbol candidate,
            INamedTypeSymbol forwardedTo,
```

Or something similar, just to make this a bit less confusing?

Reply:
(I'll say though the XML docs did make it clear so good job on that!)

Reply:
I far prefer your names.  i couldn't come up with anything good, but yours work awsome.